### PR TITLE
feat: implement M7 CLI tool

### PR DESCRIPTION
**Pull request by rsn-opencode**
## Summary

- Add argparse-based CLI (`agentguard` command) with five subcommands: `check`, `audit`, `report`, `policies`, `version`
- Zero external dependencies — uses only stdlib `argparse` for CLI parsing
- Includes `python -m agentguard` support via `__main__.py` and `pyproject.toml` console_scripts entry point

## Subcommands

| Command | Description |
|---------|-------------|
| `agentguard check` | Validate actions against policies (supports `--policy`, `--action`, `--actor`, `--resource`, `--context`) |
| `agentguard audit` | Query and verify audit logs (`query` and `verify` sub-subcommands) |
| `agentguard report` | Generate EU AI Act compliance reports (`--format json\|text`, `--output`) |
| `agentguard policies` | List and inspect built-in policies (`list` and `show` sub-subcommands) |
| `agentguard version` | Display version information |

## Testing

- 31 new tests in `tests/unit/test_cli.py` covering all subcommands and edge cases
- All 292 tests pass across Python 3.10–3.13
- ruff clean, mypy clean

## Milestone

Closes M7 (CLI tool) from the project roadmap.

### CODE REVIEW

File: src/agentguard/cli.py [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+    if args.command == "version":
+        return _cmd_version()
+
+    if args.command == "policies":
+        if args.policies_command == "list":
+            return _cmd_policies_list()
+        if args.policies_command == "show":
+            return _cmd_policies_show(args.name)
+        parser.parse_args(["policies", "--help"])
+        return 1
```
[the remark this answers is not in the document]
Comment (rsn-opencode) [reply src/agentguard/cli.py]: Fixed: Replaced `parser.parse_args(["policies", "--help"])` with `_parsers.policies.print_help()` which shows the policies-specific help and allows `return 1` to execute.

File: src/agentguard/cli.py [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+    if args.policy_dir:
+        policy_dir = Path(args.policy_dir)
+        if not policy_dir.is_dir():
+            print(
+                f"Error: Policy directory not found: {args.policy_dir}",
+                file=sys.stderr,
+            )
+            return 1
+        for yaml_file in sorted(policy_dir.glob("*.yaml")):
+            guard.load_policy_file(yaml_file)
```
[the remark this answers is not in the document]
Comment (rsn-opencode) [reply src/agentguard/cli.py]: Fixed: Added `try/except Exception` around `guard.load_policy_file()` in the `--policy-dir` loop, consistent with the `--policy` path. Using broad `Exception` to catch YAML parse errors (`yaml.scanner.ScannerError`), `ValueError`, `FileNotFoundError`, etc.

File: src/agentguard/cli.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+def _parse_params(params: list[str]) -> dict[str, str]:
+    """Parse key=value parameter strings into a dict."""
+    result: dict[str, str] = {}
+    for param in params:
+        if "=" not in param:
+            print(
+                f"Error: Invalid parameter '{param}'. Expected key=value format.",
+                file=sys.stderr,
+            )
+            continue
```
[the remark this answers is not in the document]
Comment (rsn-opencode) [reply src/agentguard/cli.py]: Fixed: `_parse_params` now raises `ValueError` on malformed params instead of silently continuing. `_cmd_check` catches the error and exits with code 2.
